src/style-property-map-readonly.js

```diff
@@ -14,27 +14,26 @@
 
 (function(shared, scope, testing) {
 
-  function ComputedStylePropertyMap(element) {
-    this._element = element;
+  function StylePropertyMapReadOnly(styleObject) {
+    this._styleObject = styleObject;
   }
 
-  ComputedStylePropertyMap.prototype =
-    Object.create(shared.StylePropertyMap.prototype);
-
-  ComputedStylePropertyMap.prototype.append = function(property, value) {
-    throw new TypeError('ComputedStylePropertyMap is immutable');
-  };
-
-  ComputedStylePropertyMap.prototype.delete = function(property) {
-    throw new TypeError('ComputedStylePropertyMap is immutable');
+  StylePropertyMapReadOnly.prototype = {
+    append: function(property, value) {},
```

Comment:
This wasn't quite what I meant. The StylePropertyMapReadOnly should not have any of the mutating methods. 
append, delete and set should all be moved into StylePropertyMap, and this map can be removed.